src/serac/physics/utilities/functional/polynomials.hpp

```diff
@@ -247,94 +249,72 @@ constexpr tensor<T, n> GaussLobattoInterpolationDerivative01([[maybe_unused]] T
+tensor<T, n> GaussLegendreInterpolation([[maybe_unused]] T x)
 {
-  if constexpr (n == 1) {
-    return {1.0};
-  }
-
-  if constexpr (n == 2) {
-    return {1.36602540378444 - 1.73205080756888 * x, -0.366025403784439 + 1.73205080756888 * x};
-  }
-
-  if constexpr (n == 3) {
-    return {1.47883055770124 + x * (-4.6243277820691 + 3.3333333333333 * x),
-            -0.66666666666667 + x * (6.6666666666667 - 6.6666666666667 * x),
-            0.187836108965431 + x * (-2.04233888459753 + 3.3333333333333 * x)};
-  }
-
+  if constexpr (n == 1) return {1};
+  if constexpr (n == 2)
+    return {1.3660254037844386467637231708 - 1.732050807568877293527446342 * x,
+            -0.3660254037844386467637231708 + 1.732050807568877293527446342 * x};
+  if constexpr (n == 3)
+    return {1.47883055770123614752987757 + x * (-4.6243277820691389617264218 + 3.33333333333333333333333333 * x),
+            -0.666666666666666666666666667 + (6.66666666666666666666666667 - 6.666666666666666666666666667 * x) * x,
+            0.1878361089654305191367891 + x * (-2.04233888459752770494024487 + 3.33333333333333333333333333 * x)};
+  if constexpr (n == 4)
+    return {
+        1.5267881254572667869843283 +
+            x * (-8.546023607872198765973019 + (14.325858354171888152966621 - 7.420540068038946105200642 * x) * x),
+        -0.8136324494869272605618981 +
+            x * (13.807166925689577066158695 + x * (-31.388222363446060212058231 + 18.795449407555060811261716 * x)),
+        0.400761520311650404800281777 + x * (-7.41707042146263907582738061 +
+                                             (24.9981258592191222217269164 - 18.79544940755506081126171563 * x) * x),
+        -0.113917196281989931222711973 +
+            x * (2.15592710364526077564170438 + x * (-7.9357618499449501626353065 + 7.4205400680389461052006424 * x))};
   return tensor<T, n>{};
 }
 
 /**
  * @brief Derivatives of the Lagrange Interpolating polynomials for nodes at Gauss-Legendre points on the interval [-1,
  * 1]
+ *
+ * Mathematica/Wolfram Language code to generate more entries in the table:
+ * Do[xi = GaussianQuadratureWeights[n, 0, 1, 30][[All, 1]];
+ *  Print["if constexpr (n == " <> ToString[n] <> ") return " <> ToString[CForm /@ HornerForm /@
+ * (D[InterpolatingPolynomial[Transpose[{xi, #}], x], x] & /@ IdentityMatrix[n])] <> ";"], {n, 1, 4}
+ * ]
+ *
+ * @tparam[in] n how many entries to compute
+ * @param[in] x where to evaluate the polynomials
  */
 template <int n, typename T>
-tensor<T, n> GaussLegendreInterpolationDerivative01(T x)
+tensor<T, n> GaussLegendreInterpolationDerivative([[maybe_unused]] T x)
 {
-  if constexpr (n == 1) {
-    return {0.0};
-  }
-
-  if constexpr (n == 2) {
-    return {-1.732050807568877, 1.732050807568877};
-  }
-
-  if constexpr (n == 3) {
-    return {-4.6243277820691 + 6.6666666666667 * x, 6.6666666666667 - 13.3333333333333 * x,
-            -2.04233888459753 + 6.6666666666667 * x};
-  }
-
+  if constexpr (n == 1) return {0};
```

Comment:
My inner CS person who hates magic numbers dies a little here... no fix is needed.